src/graphics.cpp

```diff
@@ -88,10 +88,12 @@ void Graphics::Update() {
 		if (fps_overlay->GetFps() == 0) {
 			Output::Debug("Framerate is 0 FPS!");
 			Draw();
+			Player::FrameReset();
```

Comment:
This ends up calling `DisplayUi->GetTicks()` 3 times (4?) which results in 3(4?) system calls. Even worse, it happens when your fps is 0, causing you to slow down even more when your game is already lagging.

I would change both `Player::FrameReset()` and `Graphics::FrameReset()` to optionally accept an `int32_t = 0` for the ui ticks. If it's 0, fetch call `DisplayUi->GetTicks()` to set it. 

Call `DisplayUi->GetTicks()` once at the top of `Graphics::Update()` and use it in both places it's used there. Then pass it down into the 2 `FrameReset()` calls.

Reply:
fixed this for the 0 FPS case.
There are other cases where FrameReset is called but there is no tick-variable around, so didn't change them.